[statistics] dashboard reorganization, subtitles and age distribution

skarya22 opened the pull request:
## Brief summary of changes
- Project and Visit filters were broken, they now appear
- Added a Candidate Date Registered filter 
- Made filters hidden when reloading with fresh data. It was confusing before on datasets with more data than raisinbread as it would show old data until the new stuff was loaded. This was not shown in raisinbread because it loaded so fast.
- Add ethnicity data to raisinbread as well as correct DoBs compared to their Date_registered
- Add subtitles for Panel views in the dashboard for total counts
- Added value to Pie chart tooltip from EEGNET
- Added Age Distribution chart from EEGNET
- Reorganized charts to be in better sections, can compare to live v27 on demo.loris.ca
<kbd>
<img width="970" height="504" alt="image" src="https://github.com/user-attachments/assets/15e8a716-bd1f-407c-b353-6dad8a3da7a4" />
</kbd>
<kbd>
<img width="775" height="489" alt="image" src="https://github.com/user-attachments/assets/2133a989-34a7-4745-9867-5e31b450ed59" />
</kbd>
<kbd>
<img width="946" height="506" alt="image" src="https://github.com/user-attachments/assets/1bfc6346-4d4b-4567-a2f9-57200cf27cbd" />
</kbd>


#### Testing instructions (if applicable)

1. Test every filter and confirm they show the desired behaviour
2. Add a new candidate and confirm that the date filter works
3. Create an account with less project/ site permissions but with Access Profile permission
4. See that the numbers are less in the subtitles and charts do not contain information for non user projects or sites

#### Link(s) to related issue(s)

* Resolves #9947  (Reference the issue this fixes, if any.)

jeffersoncasimir reviewed (changes requested):
The "Date Registered" filter appears broken. 

When selecting a date, the query returns 500, with the following server-side error:
`Column not found: 1054 Unknown column 'c.Date_registered' in 'where clause'`

I have confirmed that my candidate table indeed contains this column.

Also, when selecting a first filter, a seemingly inconsequential console error, that calls itself a warning, appears:
`
Warning: Cannot update a component (`StudyProgression`) while rendering a different component (`QueryChartForm`). To locate the bad setState() call inside `QueryChartForm`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render Error Component Stack
    at QueryChartForm (queryChartForm.js:13:57)


Seems to otherwise work as expected.

jeffersoncasimir commented:
Alright. In that case, I will approve this PR after the commit that fixes the testing suite

skarya22 commented:
@jeffersoncasimir apologies for commit spam! I added the age distribution chart to this PR, and also cleaned up the module. There was a lot of dead frontend code. I think this will require a re-review unfortunately. It is currently failing tests but it is unrelated, so ready for review :)

christinerogers commented:
Great work @skarya22 and @jeffersoncasimir 
Just at a glance -- Is it feasible to include the % plot display (toggle) on the Age at recruitment spline chart?  
That was the most valuable display for this data, is what we heard from researchers from EEGNet. 
They found it more useful to see what % of the cohort was (e.g.) 21 years old, not necessarily the exact number.  Let's include both if it's not a big challenge.

christinerogers commented:
Just noticed 2 further things:
- the project total in the title bar is smaller and less readable than the data in the plots.  Alan wanted this to be a headline, please take a stab at making this more visible - e.g the white background did the trick on EEGNet, though its aesthetic was too clunky otherwise.
-  Size (GB/TB) of the data collection was also a key headline / feature.  Could you address in the PR description why/when this will / won't be included in the plots?

skarya22 commented:
> @skarya22 could you mention in your PR description that this will be followed by Jefferson's PR with more EEGNet dashboard features -- e.g Examples missing from this one : my last comment plus
> 
> * the project total in the title bar is smaller and less readable than the data in the plots.  Alan wanted this to be a headline, please take a stab at making this more visible - e.g the white background did the trick on EEGNet, though its aesthetic was too clunky otherwise.
> * Size (GB/TB) of the data collection was also a key headline / feature.  Could you address in the PR description why/when this will / won't be included in the visible plots?  @skarya22 thanks

@christinerogers There are other issues for the other plots such as the size (GB/TB) that you mentioned which is assigned to @jeffersoncasimir #9944. Didn't think it was a good idea to include it all in one PR as this PR is already quite large.

christinerogers commented:
@skarya22 not sure if Jefferson passed this feedback on earlier in September:
ideally the bar/pie selector is not below the title, but out to the right side (see EEGNet's age plot).
For a few reasons:
a.  it takes up vertical real estate where it currently is. 
b.  the dashboard plots are provided partly so ppl can take screenshots and use them in papers/posters - if you move it a little more out of the way it just looks better, especially for this purpose. 
c. it's not a proper subtitle in a graph anyway

Is this possible to do?  thanks

driusan commented on modules/statistics/locale/ja/LC_MESSAGES/statistics.json:
```diff
@@ -0,0 +1,12 @@
+{
```
This file shouldn't be added, it's generated by `make`
